storage/tool/src/main/java/com/hortonworks/streamline/storage/tool/SchemaMigrationHelper.java

```diff
@@ -0,0 +1,111 @@
+package com.hortonworks.streamline.storage.tool;
+
+import com.hortonworks.streamline.storage.tool.exception.SchemaMigrationException;
+import org.flywaydb.core.Flyway;
+import org.flywaydb.core.internal.dbsupport.FlywaySqlException;
+
+import static org.flywaydb.core.internal.info.MigrationInfoDumper.dumpToAsciiTable;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class SchemaMigrationHelper {
+
+    private Flyway flyway;
+
+    public SchemaMigrationHelper(Flyway flyway) {
+        this.flyway = flyway;
+    }
+
+    private void create() throws SQLException {
+        if(!validateDatabase(true))
+            throw new SchemaMigrationException("Target database is not empty, please consider executing with \"migrate\" option");
+        flyway.migrate();
+    }
+
+    private void migrate() throws SQLException {
+        if(!validateDatabase(false))
+            throw new SchemaMigrationException("Target database is empty, please consider executing with \"create\" option");
```

Comment:
Target database is empty and no previous versions found, use the "create" option.